Helper/DeletedProductSerializer.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace Metrilo\Analytics\Helper;
+
+class DeletedProductSerializer extends \Magento\Framework\App\Helper\AbstractHelper
+{
+    private function searchForParentById($parentId, $deletedProducts) {
+        foreach ($deletedProducts as $index => $product) {
+            if ($product['item_id'] === $parentId) {
+                return $deletedProducts[$index];
+            }
+        }
+        return false;
+    }
+
+    public function serialize($deletedProducts)
+    {
+        $productBatch = [];
+        foreach ($deletedProducts as $product) {
+            $productOptions = [];
+            
+            if ($product['product_type'] == 'configurable') {
```

Comment:
Can't you skip configurables at the query level?

Reply:
I can skip them but the way Magento is saving product data in orders will leave us with no price for the child product (the logic is opposite in the product import where configurable product have a price 0 and the price is assigned to the child product, in orders its the opposite way - childs have 0 price and the amount of the variation is placed on configurable parent product in order records).

Reply:
But you're skipping them here? How do you get the price then?

Reply:
Query to the DB returns array of simple and parent - child products from orders. Then i am looping through to extract every simple product (no need for configurables since they are in that array only for the price value and if I skip them on query level child products will have 0 as price in the order records). Next i am performing a check for every simple product by taking product_item_id value (if its null, then product is simple. if it's id i am searching for that id in the array and get the parent data to create the serialisation for the sync).

Reply:
What if the parent and the child are not in the same batch?

Reply:
All deleted product data is one object, batch split on chunks happens after serialisation.

Reply:
ok